stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt

```diff
@@ -55,6 +61,31 @@ class ChatFragment : Fragment() {
         initMessagesViewModel()
         initMessageInputViewModel()
         configureBackButtonHandling()
+
+        messageListViewModel.state.observe(viewLifecycleOwner) { state ->
+            if (state is MessageListViewModel.State.Result && state.messageListItem.isThread) {
+                binding.messagesHeaderView.setThreadSubtitle(
+                    threadSubtitle(requireContext(), state.messageListItem)
+                )
+            }
+        }
+    }
+
+    private fun threadSubtitle(context: Context, messageWrapper: MessageListItemWrapper): String {
+        val users = threadUsers(messageWrapper)
+
+        return if (users.size == 1) {
+            String.format(context.getString(R.string.stream_ui_subtitle_thread_reply_single_user), users[0].name)
+        } else {
+            String.format(context.getString(R.string.stream_ui_subtitle_thread_reply_many_users), users.size)
+        }
+    }
+
+    private fun threadUsers(messageWrapper: MessageListItemWrapper): List<User> {
+        return messageWrapper.items
+            .filterIsInstance<MessageListItem.MessageItem>()
+            .filter { message -> message.isTheirs }
+            .map { messageItem -> messageItem.message.user }
```

Comment:
Won't this result in 3 of the same user being returned if they've added 3 messages to the thread? Feels like something like a `.distinct()` or `toSet()` call is missing here.